corehq/apps/reports/dispatcher.py

```diff
@@ -153,6 +153,15 @@ def dispatch(self, request, domain=None, report_slug=None, render_as=None,
                 report.decorator_dispatcher(
                     request, domain=domain, report_slug=report_slug, *args, **kwargs
                 )
+                # grep help:
+                # return report.json_response
+                # return report.async_response
+                # return report.filters_response
+                # return report.export_response
+                # return report.mobile_response
+                # return report.email_response
+                # return report.partial_response
+                # return report.print_response
```

Comment:
curious to what our recommended replacement should be for such `attr` calls?
Should this instead be replaced with something like
```
if render_as == 'json':
    return report.json_response
elif render_as == 'async':
    return report.async_response
    # ...
```
or you could instead do
```
report.get_response(render_as)
```
and have report.get_response do the attr call, which I believe will be a touch better or do the if-else chain.

sounds like something book-club would be interested to weigh in on.

Reply:
again, this documentation would not really help. People who use IDE wont benefit from this, and searching for say "json_response" would lead to a lot of matches in the code.

Reply:
Can you elaborate on why people who use an IDE wouldn't benefit? 

If I'm searching where `export_response` is called in the code, I would do something like: 
![Screenshot from 2022-01-25 10-13-40](https://user-images.githubusercontent.com/146896/151003881-b3ba4a70-d76d-4b77-8d2e-7a742bc3515d.png)

Prior to this change, when you search for that, it looks like the only place that method is called is in `reports/views.py`. This is very confusing, because that isn't where most calls to that method come from, and the only way I was able to figure it out was to put in a breakpoint in the method and then search up the call stack. 

Does your IDE know that this method is called here? If so, that's awesome - however, I don't think adding this line will hinder that usecase?

Reply:
in IDE, I'd search to find "usages" of a method which the IDE provides which isn't same as text search but actual code calls. The change here seems beneficial for if you were doing text search and that too for "export_response" but if you search for "json_response", I am guessing you would get quite some irrelevant matches.

> Does your IDE know that this method is called here? If so, that's awesome

It can't find indirect calls like this.

I am not saying that there isn't currently a problem. I am just saying that the fix here isn't able to really solve it.

Reply:
OK... But do you think this is at least slightly better than not having these comments?

Reply:
Sure yea. 

My concern is that adding these comments will make us think the problem is solved and accept this as the solution which I think isn't a good path to go down to.

Reply:
This behavior has long annoyed me.  To @mkangia 's first point, another approach would be:
```python
return {
    'json': report.json_response,
    'async': report.async_response,
    ...
}[render_as]
```
But those are `@property`s that do work, a lot of it (another antipattern), and this approach would evaluate them all.  You could hack around this like so:
```python
return {
    'json': lambda: report.json_response,
    'async': lambda: report.async_response,
    ...
}[render_as]()
```
This would work too, but it's only marginally more discoverable than the original version:
```python
response_method = {
    'json': `json_response`,
    'async': `async_response`,
    ...
}[render_as]
return getattr(report, response_method)
```

Reply:
My vote would be for Option Two: the dictionary with lambdas.

* It addresses the same problem that the grep help is addressing (although it does miss "return")
* It works for finding usages in PyCharm and VS Code
* Although the lambdas aren't ideal, it feels more concise than a 16-line `if ... elif` block, but still readable.
* Devs will be familiar with the dict lookup from other places in the codebase.